packages/engineer/src/feature/dev-server.dev-server.env.ts

```diff
@@ -30,7 +30,15 @@ interface WebpackDevMiddleware extends express.Handler {
 
 const attachWSHost = (socketServer: io.Server, envName: string, communication: Communication) => {
     const host = new WsServerHost(socketServer.of(`/${envName}`));
-
+    host.addEventListener('message', ({ data: { from, origin } }) => {
+        // we map both the from and the to, because we change mapping in the host itself, it re-mapps both the origin and the from, for multi-tenancy
+        if (!communication.getEnvironmentHost(from)) {
+            communication.registerEnv(from, host);
```

Comment:
on every message we register environment? that seems wrong

Reply:
nope.. on every message to that host we check if it's not registered. it should happen only once

Reply:
removed :)